pkg/controller/directvolumemigration/pvcs.go

```diff
@@ -72,7 +78,47 @@ func (t *Task) createDestinationPVCs() error {
 	return nil
 }
 
-func (t *Task) getDestinationPVCs() error {
+func (t *Task) areDestinationPVCsBound() (bool, error) {
 	// Ensure PVCs are bound and not in pending state
-	return nil
+	// Get client for destination
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return false, err
+	}
+
+	pvcMap := t.getPVCNamespaceMap()
+	selector := labels.SelectorFromSet(t.buildDVMLabels())
+	for ns, _ := range pvcMap {
+		pvcList := corev1.PersistentVolumeClaimList{}
+		err = destClient.List(
+			context.TODO(),
+			&k8sclient.ListOptions{
+				Namespace:     ns,
+				LabelSelector: selector,
+			},
+			&pvcList)
+		if err != nil {
+			return false, err
+		}
+		for _, pvc := range pvcList.Items {
+			if pvc.Status.Phase != corev1.ClaimBound {
+				if time.Now().Sub(pvc.CreationTimestamp.Time) > time.Minute*10 {
+					t.Owner.Status.SetCondition(migapi.Condition{
+						Type:     PVCNotBoundOnDestinationCluster,
+						Status:   True,
+						Reason:   migapi.NotReady,
+						Category: Critical,
+						Message:  fmt.Sprintf("PVC %s of %s namespace is in %s state", pvc.Name, pvc.Namespace, pvc.Status.Phase),
+						Durable:  true,
+					})
+				}
+				return false, nil
+			}
+		}
+	}
+	conditions := t.Owner.Status.Conditions.FindConditionByCategory(Critical)
+	if len(conditions) > 0 {
+		t.Owner.Status.DeleteCondition(Critical)
```

Comment:
NIT: should we delete only the condition whose type is `PVCNotBoundOnDestinationCluster` ? to avoid deleting other Critical conditions by mistake?